sphinx/builders/__init__.py

```diff
@@ -59,33 +59,38 @@ class Builder:
     Builds target formats from the reST sources.
     """
 
-    #: The builder's name, for the -b command line option.
-    name = ''
+    #: The builder's name. This is the name used for the
+    #: :option:`sphinx-build -b` command line option.
+    name: str = ''
     #: The builder's output format, or '' if no document output is produced.
-    format = ''
+    #: This is commonly the file extension, e.g. "html", but can technically
+    #: be any string. Its value can be used by various components such as
+    #: :class:`.SphinxPostTransform` or extensions to determine whether they
+    #: can work with the builder.
+    format: str = ''
     #: The message emitted upon successful build completion. This can be a
     #: printf-style template string with the following keys: ``outdir``,
     #: ``project``
-    epilog = ''
+    epilog: str = ''
 
     #: default translator class for the builder.  This can be overridden by
     #: :py:meth:`~sphinx.application.Sphinx.set_translator`.
     default_translator_class: type[nodes.NodeVisitor]
     # doctree versioning method
     versioning_method = 'none'
     versioning_compare = False
-    #: allow parallel write_doc() calls
-    allow_parallel = False
+    #: Whether it is safe to make parallel :meth:`~.Builder.write_doc()` calls.
+    allow_parallel: bool = False
```

Comment:
you don't need to set the type here, it is auto-inferred, you only need to set the type if it can be a union or has undefined generics (like an empty list)

Reply:
`.. autoattribute::` does not infer it. Compare:

- https://sphinx--13008.org.readthedocs.build/en/13008/extdev/builderapi.html#sphinx.builders.Builder.allow_parallel
- https://www.sphinx-doc.org/en/master/extdev/builderapi.html#sphinx.builders.Builder.allow_parallel

I thought it may be helpful to have the type in the documentation explicitly.

Reply:
yeh eurgh, that should really be fixed, rather than having to adapt all the documentation to deal with a bug.
if you want to type them, then you also need to use https://docs.python.org/3/library/typing.html#typing.ClassVar

Reply:
> The worst offender though is `class_var = None`, as then you've no hope of knowing the real type!

Yep that's what I meant by "if it can be a union" 👍